File: c_files/raylib/simple_platformer/player.c
#include "player.h"
#include "raylib.h"

void initPlayer(Player *player) {
  player->width = 30;
  player->height = 30;
  player->x = 100;
  player->y = 200;
  player->color = BLUE;
  player->change_x = 8;
  player->change_y = 0;
  player->gravity = 1;
  player->jump_force = -20;
  player->can_jump = true;
}

void drawPlayer(Player *player) {
  DrawRectangle(player->x, player->y, player->width, player->height,
                player->color);
}

void updatePlayer(Player *player) {
  // horizontal movement
  if (IsKeyDown(KEY_RIGHT)) {
    player->x += player->change_x;
  } else if (IsKeyDown(KEY_LEFT)) {
    player->x -= player->change_x;
  }

  // vertical movement
  if (IsKeyPressed(KEY_UP) && player->can_jump) {
    player->change_y = player->jump_force;
    player->can_jump = false;
  }

  player->change_y += player->gravity;
  player->y += player->change_y;
}

File: python_files/raylib/simple_platformer/player.py
import pyray as p

class Player:
    def __init__(self, spritesheet, sprite_dict) -> None:
        self.spritesheet = spritesheet
        self.sprite_dict = sprite_dict

        self.player_idle = self.sprite_dict["tile_0045.png"]
        
        self.x_window = 20
        self.y_window = p.get_screen_height() - 95
        self.change_x = 5
        self.change_y = 0
        self.gravity = 1
        self.jump_force = -20
        self.can_jump = True
        
        self.current_frame = self.player_idle
        self.current_frame_width = self.current_frame[2]
        self.current_frame_height = self.current_frame[3]

    def draw(self):
        p.draw_texture_pro(self.spritesheet,
                           (self.current_frame[0], self.current_frame[1], self.current_frame_width, self.current_frame_height),
                           (self.x_window, self.y_window, self.current_frame_width * 3, self.current_frame_height * 3),
                           (self.current_frame_width/2, self.current_frame_height/2),
                           0,
                           p.WHITE)
        
    def update(self):
        # horizontal movement
        if p.is_key_down(p.KeyboardKey.KEY_RIGHT):
            self.x_window += self.change_x
        elif p.is_key_down(p.KeyboardKey.KEY_LEFT):
            self.x_window -= self.change_x
            
        # vertical movement
        if p.is_key_pressed(p.KeyboardKey.KEY_SPACE) and self.can_jump:
            self.change_y = self.jump_force
            self.can_jump = False
        
        self.change_y += self.gravity
        self.y_window += self.change_y
        
